packages/camera/camera_avfoundation/ios/camera_avfoundation/Sources/camera_avfoundation/FLTCam.m

```diff
@@ -211,20 +211,50 @@ - (instancetype)initWithMediaSettings:(FCPPlatformMediaSettings *)mediaSettings
   [_motionManager startAccelerometerUpdates];
 
   if (_mediaSettings.framesPerSecond) {
+    [_mediaSettingsAVWrapper beginConfigurationForSession:_videoCaptureSession];
+
+    // Possible values for presets are hard-coded in FLT interface having
+    // corresponding AVCaptureSessionPreset counterparts.
+    // If _resolutionPreset is not supported by camera there is
+    // fallback to lower resolution presets.
+    // If none can be selected there is error condition.
+    if (![self setCaptureSessionPreset:_mediaSettings.resolutionPreset withError:error]) {
+      [_videoCaptureSession commitConfiguration];
+      return nil;
+    }
+
     // The frame rate can be changed only on a locked for configuration device.
     if ([mediaSettingsAVWrapper lockDevice:_captureDevice error:error]) {
-      [_mediaSettingsAVWrapper beginConfigurationForSession:_videoCaptureSession];
-
-      // Possible values for presets are hard-coded in FLT interface having
-      // corresponding AVCaptureSessionPreset counterparts.
-      // If _resolutionPreset is not supported by camera there is
-      // fallback to lower resolution presets.
-      // If none can be selected there is error condition.
-      if (![self setCaptureSessionPreset:_mediaSettings.resolutionPreset withError:error]) {
-        [_videoCaptureSession commitConfiguration];
-        [_captureDevice unlockForConfiguration];
-        return nil;
+      // find the format which frame rate ranges are closest to the wanted frame rate
```

Comment:
Comments need to follow standard sentence rules. https://google.github.io/styleguide/objcguide.html#comments